tests/unit/backend/corpora/common/entities/test_project.py

```diff
@@ -118,6 +120,52 @@ def test__create_ids__ok(self):
         self.assertEqual(link_ids, get_ids(project.links))
         self.assertNotEqual(["test_project_link_id"], get_ids(project.links))
 
+    def test__list_in_time_range__ok(self):
```

Comment:
As this test is written today, all the projects that are created are returned no matter what the time range is. While you do check to make sure that all the returned projects have a timestamp that is acceptable, you do not check to make sure that certain projects were _excluded_. i.e. you check for true positives but not true negatives in confusion matrix terms.

Here's a way you could potentially frame this test. Create 3 different projects while mocking out the timestamp such that they are created at different times. Control the timestamps yourself. Then conduct the tests such that you have your permutations of the parametrization of the function, but again, control the from and to dates such that certain projects should be excluded each time (or not for the case where you do not parametrize the function). You can check the timestamps of the project but you can also check which projects are included and which ones are excluded explicitly because you controlled the timestamp when you created the projects.

Alternatively, you'd have one test per parametrization of the function where you only need to generate two projects at a time. This would be my preferred route because it tests one functionality at a time, but I know you like subtests so suggesting the former ;)